scripts/main/upload.js

```diff
@@ -520,12 +520,13 @@ upload.start = {
 
 		/** @param {ServerImportDialogResult} data */
 		const action = function (data) {
-			if (!data.path.trim()) {
-				basicModal.error("path");
+			if (!data.paths.trim()) {
+				basicModal.error("paths");
 				return;
 			} else {
 				// Consolidate `data` before we close the modal dialog
 				// TODO: We should fix the modal dialog to properly return the values of all input fields, incl. check boxes
+				data.paths = data.paths.match(/(?:\\.|\S)+/g);
```

Comment:
Why this complicated Rexex? We want to split at every space except for spaces which are escaped by `\`, right? So why do we need the question mark and the colon? Shouldn't `/(\\ |\S+)+/g` be sufficient? I haven't tested it yet.

In the long run I would rather prefer to have a model dialog which dynamically creates another input field when the existing input fields are non-empty and which removes an input field again, if its input is deleted. This means there should always be a list of input field with one empty input field at the bottom. This way we could avoid using fragile regular expressions at all.

Reply:
> why do we need the question mark and the colon?

`( ... )` defines a capturing section
`(?: ... )` defines a non capturing section.

Reply:
I haven't the time to test it and obviously I am too tired to grasp the correctness of this expression. I assume you have tested it and it does what it is supposed to do? For me it still looks wrong even after I know that `(?: ...) defines a non-capturing group.

Reply:
https://stackoverflow.com/a/66319988/3981163